app/app_client_test.go

```diff
@@ -1523,42 +1522,21 @@ func TestAppClient(t *testing.T) {
 				NextPageToken: pageToken,
 			}, nil
 		}
-		resp, token, err := client.GetFragmentHistory(context.Background(), fragmentID, GetFragmentHistoryOptions{&pageToken, &int64Limit})
+		resp, token, err := client.GetFragmentHistory(context.Background(), fragmentID, &GetFragmentHistoryOptions{&pageToken, &int64Limit})
 		test.That(t, err, test.ShouldBeNil)
 		test.That(t, token, test.ShouldEqual, pageToken)
 		test.That(t, resp, test.ShouldResemble, expectedHistory)
 	})
 
 	t.Run("createAuthorization", func(t *testing.T) {
-		resp, err := createAuthorization(authorization.OrganizationID,
+		resp := createAuthorization(authorization.OrganizationID,
 			authorization.IdentityID,
 			authorization.IdentityType,
-			authorization.AuthorizationType,
-			authorization.ResourceType,
+			AuthRole(authorization.AuthorizationType),
+			AuthResourceType(authorization.ResourceType),
```

Comment:
This looks a bit strange to me. why is `authorization.AuthorizationType` still a string? I suspect we should update the `Authorization` struct to expect a more typeful `AuthorizationType` and `ResourceType`